base/applications/rapps/appview.cpp

```diff
@@ -1382,7 +1382,33 @@ BOOL CAppsListView::AddInstalledApplication(CInstalledApplicationInfo *InstAppIn
         return FALSE;
     }
 
-    HICON hIcon = (HICON)LoadIconW(hInst, MAKEINTRESOURCEW(IDI_MAIN));
+    /* Load icon from registry */
+    HICON hIcon = NULL;
+    ATL::CStringW szIconPath;
+    if (InstAppInfo->RetrieveIcon(szIconPath))
+    {
+        INT Ret = PathParseIconLocationW((LPWSTR)szIconPath.GetString());
+
+        /* Check if icon location is zero-terminated */
+        if (Ret == 0)
```

Comment:
I don't understand this (and the "else if (!Ret)" below).
You are basically doing this:
```
if (Ret == 0)
{
    hIcon = ExtractIcon(..., Ret /* which is == 0 */);
}
else if (Ret != 0) // Note that just using "else" would have been sufficient...
{
    hIcon = ExtractIcon(..., 0);
}
```
In other words in both cases, you did the very same call `hIcon = ExtractIcon(..., 0);`
Did you want instead to use the actual returned icon index in the ExtractIcon call (and not just always '0')? If so you don't need to use all this complicated ifs, but just need to use `hIcon = ExtractIcon(..., Ret);` .
And instead, if you always want to use icon 0, just ignore the returned value of the PathParseIconLocationW() call and directly call `hIcon = ExtractIcon(..., 0);` .

Reply:
Fixed. Thanks for the tip @HBelusca. :+1: 
It still displays all icons with the first way: just use return value from `PathParseIconLocationW` and don't do any checks of it.
But the 2nd way proposed by you is not suitable in this case, because without parsing the string from registry, it displays only the icons from strings without icon index at the end (as I already said in the PR description).

Reply:
>  But the 2nd way proposed by you is not suitable in this case, because without parsing the string from registry, it displays only the icons from strings without icon index at the end (as I already said in the PR description).

In the 2nd way I still was proposing to use `PathParseIconLocationW()`, but to not take care about the returned value that you store into Ret; in other words the 2nd way would have been just:
```c
PathParseIconLocationW((LPWSTR)szIconPath.GetString());

hIcon = ExtractIconW(hInst,
                     szIconPath.GetString(),
                     0);
```
(So, this would have been only if you wanted to be able to parse these registry strings with the index appended, but not take into account that index and use instead always zero.)